Assets/Mirror/Components/Experimental/NetworkTransformBase.cs

```diff
@@ -0,0 +1,449 @@
+        void SetGoal(Vector3 position, Quaternion rotation, Vector3 scale)
+        {
+            // put it into a data point immediately
+            DataPoint temp = new DataPoint
+            {
+                // deserialize position
+                localPosition = position,
+                localRotation = rotation,
+                localScale = scale,
+                timeStamp = Time.time,
+            };
+
+            // movement speed: based on how far it moved since last time has to be calculated before 'start' is overwritten
+            temp.movementSpeed = EstimateMovementSpeed(goal, temp, targetComponent.transform, syncInterval);
+
+            // reassign start wisely
+            // first ever data point? then make something up for previous one so that we can start interpolation without waiting for next.
+            if (start.timeStamp == 0)
+            {
+                start = new DataPoint
+                {
+                    timeStamp = Time.time - syncInterval,
+                    // local position/rotation for VR support
+                    localPosition = targetComponent.transform.localPosition,
+                    localRotation = targetComponent.transform.localRotation,
+                    localScale = targetComponent.transform.localScale,
+                    movementSpeed = temp.movementSpeed
+                };
+            }
+            // second or nth data point? then update previous
+            // but: we start at where ever we are right now, so that it's perfectly smooth and we don't jump anywhere
+            //
+            //    example if we are at 'x':
+            //
+            //        A--x->B
+            //
+            //    and then receive a new point C:
+            //
+            //        A--x--B
+            //              |
+            //              |
+            //              C
+            //
+            //    then we don't want to just jump to B and start interpolation:
+            //
+            //              x
+            //              |
+            //              |
+            //              C
+            //
+            //    we stay at 'x' and interpolate from there to C:
+            //
+            //           x..B
+            //            \ .
+            //             \.
+            //              C
+            //
+            else
+            {
+                float oldDistance = Vector3.Distance(start.localPosition, goal.localPosition);
+                float newDistance = Vector3.Distance(goal.localPosition, temp.localPosition);
+
+                start = goal;
+
+                // teleport / lag / obstacle detection: only continue at current position if we aren't too far away
+                // local position/rotation for VR support
+                if (Vector3.Distance(targetComponent.transform.localPosition, start.localPosition) < oldDistance + newDistance)
```

Comment:
```suggestion
                // XC  < AB + BC (see comments above)
                if (Vector3.Distance(targetComponent.transform.localPosition, start.localPosition) < oldDistance + newDistance)
```

Reply:
Well github made quite the mess of "insert 1 comment line before the if"